From 5c77d25134aba12866b67e29780a62fe63664b00 Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Wed, 10 Sep 2014 18:29:02 -0400
Subject: [ethernet] ixgbe: never generate both software and hardware timestamps

Message-id: <1410373750-11341-43-git-send-email-jogreene@redhat.com>
Patchwork-id: 93558
O-Subject: [RHEL7.1 PATCH 42/50] ixgbe: never generate both software and hardware timestamps
Bugzilla: 1091123
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>

BZ: 1091123
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7919168
Tested: In beaker. See BZ for details

skb_tx_timestamp() does not report software time stamp
if SKBTX_IN_PROGRESS is set. According to timestamping.txt
software time stamps are a fallback and should not be
generated if hardware time stamp is provided.

Move call to skb_tx_timestamp() after setting
SKBTX_IN_PROGRESS.

Signed-off-by: Jakub Kicinski <kubakici@wp.pl>
Tested-by: Phil Schmitt <phillip.j.schmitt@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit ff29a86ec9143ff1aa1901ae60f06f4980981df4)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
index 7b7cf80..d67ff24 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
@@ -6778,8 +6778,6 @@ netdev_tx_t ixgbe_xmit_frame_ring(struct sk_buff *skb,
   tx_flags |= IXGBE_TX_FLAGS_SW_VLAN;
  }
 
- skb_tx_timestamp(skb);
-
  if (unlikely(skb_shinfo(skb)->tx_flags & SKBTX_HW_TSTAMP)) {
   skb_shinfo(skb)->tx_flags |= SKBTX_IN_PROGRESS;
   tx_flags |= IXGBE_TX_FLAGS_TSTAMP;
@@ -6790,6 +6788,8 @@ netdev_tx_t ixgbe_xmit_frame_ring(struct sk_buff *skb,
   schedule_work(&adapter->ptp_tx_work);
  }
 
+ skb_tx_timestamp(skb);
+
 #ifdef CONFIG_PCI_IOV
  /*
   * Use the l2switch_enable flag - would be false if the DMA
-- 
1.7.1